catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/SourceOperations.java

```diff
@@ -201,22 +207,29 @@ public SourceInfoResponse getSourceInfo(
       if (sourceInfoRequest.isEnterprise()) {
 
         sourceDescriptors =
-            getFederatedSourceDescriptors(frameworkProperties.getFederatedSources().values(), true);
+            getFederatedSourceDescriptors(frameworkProperties.getFederatedSources(), true);
         // If Ids are specified check if they are known sources
       } else if (requestedSourceIds != null) {
         LOGGER.debug("getSourceRequest contains requested source ids");
         Set<FederatedSource> discoveredSources = new HashSet<>();
-        boolean containsId = false;
 
         for (String requestedSourceId : requestedSourceIds) {
           // Check if the requestedSourceId can be found in the known federatedSources
 
-          if (frameworkProperties.getFederatedSources().containsKey(requestedSourceId)) {
-            containsId = true;
-            LOGGER.debug("Found federated source: {}", requestedSourceId);
-            discoveredSources.add(frameworkProperties.getFederatedSources().get(requestedSourceId));
-          }
-          if (!containsId) {
+          final Optional<FederatedSource> source =
+              frameworkProperties
+                  .getFederatedSources()
+                  .stream()
+                  .filter(e -> e.getId().equals(requestedSourceId))
+                  .findFirst();
```

Comment:
❓ Is there a risk of there being more than one source with the `requestedSourceId`? Should this case be handled (e.g. log something)?

Reply:
Good question, I originally had a list and was always getting the first element and thought this looked cleaner. If the ID isn't a unique key then this is definitely a possibility that could happen but I'm not entirely sure how we'd want to handle that.

Reply:
It is expected but not guaranteed/enforced to be unique.